client/history/client_test.go

```diff
@@ -0,0 +1,866 @@
+						TaskToken: []byte(`{"workflowId": "test-workflow"}`),
+					},
+				})
+			},
+			mock: func(p *MockPeerResolver, c *MockClient) {
+				p.EXPECT().FromWorkflowID("test-workflow").Return("test-peer", nil).Times(1)
+				c.EXPECT().RespondActivityTaskFailed(gomock.Any(), gomock.Any(), []yarpc.CallOption{yarpc.WithShardKey("test-peer")}).
+					Return(nil).Times(1)
+			},
+		},
+		{
+			name: "RespondDecisionTaskFailed",
+			op: func(c Client) error {
+				return c.RespondDecisionTaskFailed(context.Background(), &types.HistoryRespondDecisionTaskFailedRequest{
+					FailedRequest: &types.RespondDecisionTaskFailedRequest{
+						TaskToken: []byte(`{"workflowId": "test-workflow"}`),
+					},
+				})
+			},
+			mock: func(p *MockPeerResolver, c *MockClient) {
+				p.EXPECT().FromWorkflowID("test-workflow").Return("test-peer", nil).Times(1)
+				c.EXPECT().RespondDecisionTaskFailed(gomock.Any(), gomock.Any(), []yarpc.CallOption{yarpc.WithShardKey("test-peer")}).
+					Return(nil).Times(1)
+			},
+		},
+		{
+			name: "ScheduleDecisionTask",
+			op: func(c Client) error {
+				return c.ScheduleDecisionTask(context.Background(), &types.ScheduleDecisionTaskRequest{
+					WorkflowExecution: &types.WorkflowExecution{WorkflowID: "test-workflow"},
+				})
+			},
+			mock: func(p *MockPeerResolver, c *MockClient) {
+				p.EXPECT().FromWorkflowID("test-workflow").Return("test-peer", nil).Times(1)
+				c.EXPECT().ScheduleDecisionTask(gomock.Any(), gomock.Any(), []yarpc.CallOption{yarpc.WithShardKey("test-peer")}).
+					Return(nil).Times(1)
+			},
+		},
+		{
+			name: "SignalWorkflowExecution",
+			op: func(c Client) error {
+				return c.SignalWorkflowExecution(context.Background(), &types.HistorySignalWorkflowExecutionRequest{
+					SignalRequest: &types.SignalWorkflowExecutionRequest{
+						WorkflowExecution: &types.WorkflowExecution{WorkflowID: "test-workflow"},
+					},
+				})
+			},
+			mock: func(p *MockPeerResolver, c *MockClient) {
+				p.EXPECT().FromWorkflowID("test-workflow").Return("test-peer", nil).Times(1)
+				c.EXPECT().SignalWorkflowExecution(gomock.Any(), gomock.Any(), []yarpc.CallOption{yarpc.WithShardKey("test-peer")}).
+					Return(nil).Times(1)
+			},
+		},
+		{
+			name: "SyncActivity",
+			op: func(c Client) error {
+				return c.SyncActivity(context.Background(), &types.SyncActivityRequest{
+					WorkflowID: "test-workflow",
+				})
+			},
+			mock: func(p *MockPeerResolver, c *MockClient) {
+				p.EXPECT().FromWorkflowID("test-workflow").Return("test-peer", nil).Times(1)
+				c.EXPECT().SyncActivity(gomock.Any(), gomock.Any(), []yarpc.CallOption{yarpc.WithShardKey("test-peer")}).
+					Return(nil).Times(1)
+			},
+		},
+		{
+			name: "SyncShardStatus",
+			op: func(c Client) error {
+				return c.SyncShardStatus(context.Background(), &types.SyncShardStatusRequest{
+					ShardID: 123,
+				})
+			},
+			mock: func(p *MockPeerResolver, c *MockClient) {
+				p.EXPECT().FromShardID(123).Return("test-peer", nil).Times(1)
```

Comment:
let's add more test cases to cover
- peerResolver's corresponding method returning err causes client method to return err
- mock client's corresponding method returning err causes client method to return err

otherwise there are plenty of code branches uncovered. https://app.codecov.io/github/uber/cadence/commit/eaacb01073f6fc3c844329d557b732f3102cc41b/blob/client/history/client.go